feather.ml

```diff
@@ -30,15 +30,29 @@ type env = (string * string) list
 
 (* TODO: implement [wait] *)
 
-(* TODO: Refactor to use a record *)
+type context = {
+  stdin_reader : Unix.file_descr;
+  stdout_writer : Unix.file_descr;
+  stderr_writer : Unix.file_descr;
+  background : bool;
+  cwd : string option;
+  env : env option;
+}
+
 type cmd =
-  stdin_reader:Unix.file_descr ->
-  stdout_writer:Unix.file_descr ->
-  stderr_writer:Unix.file_descr ->
-  background:bool ->
-  cwd:string option ->
-  env:env option ->
-  unit
+  | Process of string * string list
+  | Pipe of cmd * cmd
+  | And of cmd * cmd
+  | Or of cmd * cmd
+  | Seq of cmd * cmd
+  | WOut of string * cmd
```

Comment:
Do you mind using longer names here? I find abbreviations confusing sometimes.

Reply:
No problem I can change this.